src/main/java/com/farion/onlinebookstore/validator/CategoryIdsValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.farion.onlinebookstore.validator;
+
+import com.farion.onlinebookstore.lib.CategoryIds;
+import com.farion.onlinebookstore.service.CategoryService;
+import com.farion.onlinebookstore.util.ConditionHolder;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class CategoryIdsValidator implements ConstraintValidator<CategoryIds, Set<Long>> {
+    private final ConditionHolder conditionHolder;
+    private final CategoryService categoryService;
+
+    @Override
+    public void initialize(CategoryIds constraintAnnotation) {
+        conditionHolder.setCondition(ThreadLocal.withInitial(categoryService::getAllCategoryIds));
+    }
```

Comment:
this will also most likely not work as you'd expect, because I think It'll be called only once at the start of the application, and then these id's will become outdated